kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/testplugins/ClientAuthAwareLawyerFilter.java

```diff
@@ -24,7 +24,8 @@
 import edu.umd.cs.findbugs.annotations.NonNull;
 
 /**
- * A filter that adds {@linkplain FilterContext#clientTlsContext() client-facing TLS context}-dependent headers to produced records.
+ * A filter that adds {@linkplain FilterContext#clientTlsContext() client-facing TLS context}-dependent or
+ * {@linkplain FilterContext#clientSaslContext() SASL context} headers to produced records.
```

Comment:
to me this reads a little ambiguously, like it does one or the other but not both. Maybe something more like:

```suggestion
* A filter that adds headers to produced records with information from the
* {@linkplain FilterContext#clientTlsContext() TLS} and 
* {@linkplain FilterContext#clientSaslContext() SASL} context, if present.
```